### PR TITLE
FE: Create a single instance of notification in form of Context Provider and use it across whole app

### LINKED ISSUE #2166: FE: Create a single instance of notification in form of Context Provider and use it across whole app
Let's create a single instance of notification in form of Context Provider and use it across whole app

Thanks

_Originally posted by @palashgdev in https://github.com/SigNoz/signoz/pull/2161#pullrequestreview-1280388150_

### PR DESCRIPTION
**Pull request by gitstart**
Fixes #2166 

## Demo
https://www.loom.com/share/bb4e27fa79ed4b23b8b174062b33e776

**Note**: All the errors that appear in the "Dashboard" and "Traces" tabs were forced to trigger notifications. These errors are triggered by Redux actions. "Logs" tab notifications are triggered by common components.

### CODE REVIEW

File: frontend/src/AppRoutes/Private.tsx [hunk 1]
```diff
@@ -1,13 +1,12 @@
...
-import { notification } from 'antd';
 import getLocalStorageApi from 'api/browser/localstorage/get';
 import loginApi from 'api/user/login';
 import { Logout } from 'api/utils';
 import Spinner from 'components/Spinner';
 import { LOCALSTORAGE } from 'constants/localStorage';
 import ROUTES from 'constants/routes';
 import history from 'lib/history';
-import React, { useEffect, useMemo } from 'react';
+import React, { useContext, useEffect, useMemo } from 'react';
```
Comment (palashgdev) [inline frontend/src/AppRoutes/Private.tsx:9]: rather than importing useContext everywhere can we make a custom hook ?

**Review by palashgdev — CHANGES_REQUESTED** [review]

File: frontend/src/AppRoutes/Private.tsx (hunk 1 above)
Comment (gitstart) [reply frontend/src/AppRoutes/Private.tsx:9]: Hi @palashgdev, thank you for the review. Making custom hooks makes sense, will work on the changes and update the PR

**Comment by gitstart** [comment]
Hi @palashgdev! PR has been updated, please take a look

File: frontend/src/AppRoutes/Private.tsx [hunk 2]
```diff
@@ -19,6 +18,7 @@ import { UPDATE_USER_IS_FETCH } from 'types/actions/app';
 import AppReducer from 'types/reducer/app';
 import { routePermission } from 'utils/permission';
 
+import { useNotifications } from './notifications';
```
Comment (palashgdev) [inline frontend/src/AppRoutes/Private.tsx:21]: there is a hook in src please place it there

File: frontend/src/AppRoutes/index.tsx [hunk 3]
```diff
@@ -1,40 +1,47 @@
...
-							<Switch>
-								{routes.map(({ path, component, exact }) => (
-									<Route
-										key={`${path}`}
-										exact={exact}
-										path={path}
-										component={component}
-									/>
-								))}
+			<NotificationContext.Provider value={notifications}>
```
Comment (palashgdev) [inline frontend/src/AppRoutes/index.tsx:22]: Also please make the component for this provider in the hook itself and use it as <NotificationProvider>

File: frontend/src/AppRoutes/notifications.tsx [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+import React, { Context, useContext } from 'react';
+
+type Notification = {
+	notifications: NotificationInstance;
+};
+
+export const NotificationContext: Context<Notification> = React.createContext<Notification>(
+	{
+		notifications: {
+			success: () => {},
```
Comment (palashgdev) [inline frontend/src/AppRoutes/notifications.tsx:11]: VoidFunction ?

**Review by palashgdev — CHANGES_REQUESTED** [review]

**Comment by palashgdev** [comment]
Thanks for the sharing the PR @gitstart this is a good start :-D

**Comment by gitstart** [comment]
Hi @palashgdev, requested changes have been addressed. Please review

**Review by palashgdev — APPROVED** [review]

**Pull request merged**